src/test/server/store-test.js

```diff
@@ -0,0 +1,201 @@
+const assert = require('assert');
+
+const usersData  = require('../../data/users.json');
+const UserAction = require('../../../_compiled_/actions/user-action').default;
+const UserInfo   = require('../../../_compiled_/stores/user-info').default;
+const UserStore  = require('../../../_compiled_/stores/user-store').default;
+const EVENT_TYPE = require('../../../_compiled_/stores/event-type').default;
+
+const TEST_USER = new UserInfo(
+    {
+        id:      11,
+        name:    'Ervin New',
+        phone:   '89500418181',
+        email:   'vika.chernookaya@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const TEST_USER_EXISTING = new UserInfo(
+    {
+        id:      11,
+        name:    'Ervin Howell',
+        phone:   '0418181',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_USER_INFO = new UserInfo(
+    {
+        name: 'Ervin'
+    }
+);
+
+const SEARCH_USER_INFO_2 = new UserInfo(
+    {
+        name:  'Ervin  ',
+        phone: '  '
+    }
+);
+
+const UPDATE_USER = new UserInfo(
+    {
+        id:      2,
+        name:    'Vika',
+        phone:   '555',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_FIELDS = {
+    name:    true,
+    phone:   false,
+    email:   false,
+    website: false
+};
+
+let users;
+let foundUsers;
+let searchFields;
+let userId;
+
+UserStore.on(EVENT_TYPE.usersFound, _getFoundUsers);
+UserStore.on(EVENT_TYPE.addingFailed, _getUserID);
+
+function _getFoundUsers (users, fields) {
+    foundUsers   = users;
+    searchFields = fields;
+}
+
+function _getUserID (id) {
+    userId = id;
+}
+
+function _initUserList () {
+    return usersData.map((user) => {
+        return new UserInfo(
+            {
+                id:      user.id,
+                name:    user.name,
+                phone:   user.phone,
+                email:   user.email,
+                website: user.website
+            },
+            user.address,
+            user.company
+        );
+    });
+}
+
+describe('UserStore', () => {
+    beforeEach(() => {
+        users = _initUserList();
+    });
+
+    afterEach(() => {
+        UserStore._users        = _initUserList();
+        UserStore._searchedUser = {};
+    });
+
+    describe('.getUsers()', () => {
+        it('Should return a list of users', () => {
+            assert(UserStore.getUsers(), users);
+        });
+    });
+
+    describe('Actions', () => {
+        describe('addNewUser', () => {
+            it('Should emit userAdded event', () => {
+                users.push(TEST_USER);
```

Comment:
It quite uncommon way to use one array to execute assertions throughout all tests

Reply:
In previous test we checked all users so here we can check that the last one equals with `TEST_USER`  with assigned id
and you can check userList length if you like

Reply:
`TEST_USER` should not have `id`